autoload/db.vim

```diff
@@ -620,12 +614,19 @@ endfunction
 
 function! s:vim_job.call_on_finish_if_closed() abort
   if self.close && self.exit
-    return self.on_finish(split(self.output, "\n", 1), self.exit_status)
+    let output = split(self.output, "\n", 1)
+    if len(output) && empty(output[-1])
+      call remove(output, -1)
+    endif
+    return self.on_finish(output, self.exit_status)
   endif
 endfunction
 
 function! s:nvim_job_callback(jobid, data, event) dict abort
   if a:event ==? 'exit'
+    if len(self.output) && empty(self.output[-1])
+      call remove(self.output, -1)
+    endif
```

Comment:
This is the wrong blank string to remove. The current form looks like this:

```
['stderr', '', 'stdout', '']
```

We want it to be this:

```
['stderr', 'stdout', '']
```

But instead this change produces:

```
['stderr', '', 'stdout']
```

It's got the same bogus blank line from before, but now it's also going to be `'noendofline'`.

What you want to do is remove the blank line from `a:data` when it first comes in, both times. Then add one blank line back here.

Or drop the buffering and go with the <https://github.com/neovim/neovim/blob/d367ed9b23d481998d297d812f54b950e5511c24/runtime/doc/channel.txt#L93-L100> technique I linked earlier, which would have the added advantage of mixing stdout and stderr together in the correct order.

Reply:
> Or drop the buffering and go with the [neovim/neovim@d367ed9/runtime/doc/channel.txt#L93-L100](https://github.com/neovim/neovim/blob/d367ed9b23d481998d297d812f54b950e5511c24/runtime/doc/channel.txt#L93-L100) technique I linked earlier, which would have the added advantage of mixing stdout and stderr together in the correct order.

Did this since it's the suggested way to handle it.